src/sss_client/sss_pam_compat.h

```diff
@@ -49,4 +49,15 @@
 # define PAM_BAD_ITEM PAM_USER_UNKNOWN
 #endif /* PAM_BAD_ITEM */
 
+#if !defined(PAM_NONNULL) && defined(OPENPAM_NONNULL)
+#define PAM_NONNULL OPENPAM_NONNULL
+#endif
+
+#ifndef HAVE_PAM_MODUTIL_GETLOGIN
+const char * PAM_NONNULL((1)) pam_modutil_getlogin(pam_handle_t *pamh)
+{
+    return getlogin();
```

Comment:
If I read https://github.com/Distrotech/Linux-PAM/blob/60e9641fa47bebcec10b94dee92f8257c09eb45d/libpam/pam_modutil_getlogin.c#L58 correctly, then `pam_modutil_getlogin()` returns a memory chunk from the heap that a caller should handle/dispose.
@ikerexxe, is this correct?

`getlogin()` returns a "string [that] is statically allocated and might be overwritten on subsequent calls".

Maybe wrap in `strdup()` (if non-null) to have the same semantic?

Reply:
The up-to-date source for this function seems to be https://github.com/linux-pam/linux-pam/blob/bc856cd9b9b461e8e2a537f4d9db87d315f5fe7b/libpam/pam_modutil_getlogin.c and does exactly that.

The reason I decided not to use `strdup` is https://github.com/SSSD/sssd/blob/2cf2e83a223f0eb74bd81a7fcaf726c62d577719/src/sss_client/pam_sss.c#L1449
We're putting a constant `""` string there, which shouldn't be `free()`ed. If we're going `strdup`, then this line should have it too?

Reply:
I also figured pointer is saved in `pamh`, so it likely shouldn't be freed.

Reply:
But then, there is a leak with Linux `pam_modutil_getlogin`?

Reply:
Right, whatever is allocated inside pamh is released at the end of the execution of the PAM context so you don't have to worry about doing it manually.